### ISSUE TITLE
[Steps Component] Step enhancement proposal

### ISSUE THREAD

**Issue by jforatier** [AxaGuilDEv/react-toolkit#763]
### Issue and Steps to Reproduce

- Make number mandatory - Number property is mandatory on <Step> component, it used to display number before label.
- Disable (with parameter or add property) color on last step - Toolkit set it green
- Issue with colors and disabled - Maybe make it darker when disabled ? Manage state colors with params ?

#### Versions
1.3.14

##### Expected

![image](https://user-images.githubusercontent.com/658065/96875362-6c906c00-1477-11eb-929a-2faeb2b3ffe7.png)

##### Actual
![image](https://user-images.githubusercontent.com/658065/96875971-238ce780-1478-11eb-98e2-f67b2fdc636f.png)
![image](https://user-images.githubusercontent.com/658065/96876063-40c1b600-1478-11eb-9b66-244d967431d7.png)
![image](https://user-images.githubusercontent.com/658065/96876132-59ca6700-1478-11eb-82d9-eaf46946cf43.png)

#### Additional Details

- Installed packages

**Comment by arnaudforaison** [comment]
So @Railk, steps component must have always a number or can doesn't have it ?
And last step must be green ?

**Comment by Railk** [comment]
- Numbers are mandatory on the step tracker before the name of the step, always.
- The Last step is green when you reach it (activated), the other states [of this final step] follow the same rules as the others steps

**Comment by Railk** [comment]
here's the a link with the view on sketch cloud :

- https://www.sketch.com/s/74c39b6d-9920-429b-ac77-33659f7c6d63/a/jGQ5n7
- https://www.sketch.com/s/74c39b6d-9920-429b-ac77-33659f7c6d63/a/dzm9yl
- https://www.sketch.com/s/74c39b6d-9920-429b-ac77-33659f7c6d63/a/zw3m9y

and @arnaudforaison don't hesitate to tag also David ^^

**Comment by jforatier** [comment]
My proposal is based on a more 'global' process use-case. We are talking about 'form' process step here. Maybe there there is a better component to display a such status ? Overview of item lifecycle i guess.

**Comment by Railk** [comment]
Not sure what you mean, can you explain with more context ?

All our apps use this step tracker system for subscription forms since the toolkit v1, and our UX goals is to keep the homogeneity in all the apps for our end users. And step trackers are the default pattern for forms anywhere on the web.
Did you mean something else I didn't understand ?

**Comment by djaumes** [comment]
Hi @jforatier, if I get it properly, your proposition is :
• Mandatory steps number >> OK
• Last step without a green accent >> We prefer to stay on a visible sign of confirmation, with green color and the name of the step in accordance with the ending of process (ie. "Confirmation" "Validation" instead of "Inactif")
• Darker/lighter for disabled steps >> Do you wish to change the color of past/done step darker, next steps lighter? We could make the difference clearer.
• Manage states with parameters >> On your side sure, we already have state specific properties already in Axure (Disabled, Selected, Hover, etc.) that we use extensively.

**Comment by jforatier** [comment]
👍 Got it, thanks for the reply. 
I've diverted the use of the <Steps> component

It seem i'm looking for **timeline component** for my use-case (https://material-ui.com/components/timeline/)
Is there a chance to see it appear on the [design system](https://axaguildev.github.io/design-system/) ?

**Comment by Railk** [comment]
@jforatier yes we have that in our customs components not yet added to the main library. You can contact the people who are working on the OSMOSE project and ask them about what we named, for now, vertical step tracker.
